Clarify ristretto255 and journalist fetching key usage

rocodes opened the pull request:
**Description** 

- Clarify where we use ristretto255 keys as opposed to plain curve25519. For now this avoids talking about the curve25519 keys in dh-akem; this is just about the fetching key and ephemeral clue keys. The intention is to at minimum convey our use of ristretto255 for fetching and the fetching challenges. Terminology/notation not my strong suit so please feel free to make adjustments if it's clunky (cc @redshiftzero) 

- In addition, clarify that the fetching key is signed+sent once, on enrollment, and is not part of every journalist one-time key bundle (#127)


Fixes #127 
Fixes #128

rocodes committed:
protocol: Specify Ristretto255.KGen for fetching keys and private scalar for fetch and challenges.

Add Ristretto255 operations to building blocks. Use DH operator instead of exponent notation in message sending protocol

rocodes committed:
chore: fixup backticks in protocol.md

rocodes committed:
protocol: Remove fetching key from one-time key bundles (#127)

rocodes committed:
format: lint table

cfm commented on docs/protocol.md:
```diff
@@ -91,25 +91,27 @@ Throughout this document, keys are notated as $component_{owner}^{scheme}$, wher
+|                      | $`c \gets^{\$} \text{AuthEnc}(sk, pk, m, ad, info)`$      | Encrypt a message $m$ to a recipient's public key $pk$ using private key $sk$, associated data $ad$, and $info$                                                                                                |
+|                      | $`m \gets \text{AuthDec}(sk, pk, c, ad, info)`$           | Decrypt a ciphertext $c$ using a recipient's private key $sk$ and a sender's public key $pk$; rest as above                                                                                                    |
+| [`Fetch`][fetched]   | $`(sk, pk) \gets^{\$} \text{Ristretto255.KGen}()`$        | Generate a Ristretto255 DH keypair by sampling $`x \gets^{\$} \mathbb{F}_\ell`$, the Ristretto255 scalar field, and computing $`pk = x \cdot B`$, where $`B \in \mathbb{G}_{\mathrm{R255}}`$ is the basepoint. |
+|                      | $`K \gets \mathsf{DH}(sk, pk')`$                          | Perform a Diffie--Hellman agreement between two Ristretto255 keys, outputting $`K = sk \cdot pk' = sk' \cdot pk \in \mathbb{G}_{\mathrm{R255}}`$.                                                              |
```
```suggestion
| [`Fetch`][fetched]   | $`(sk, pk) \gets^{\$} \text{KGen}()`$        | Generate a ristretto255 Diffie-Hellman keypair by sampling $`x \gets^{\$} \mathbb{F}_\ell`$, the ristretto255 scalar field, and computing $`pk = x \cdot B`$, where $`B \in \mathbb{G}_{\mathrm{R255}}`$ is the basepoint |
|                      | $`K \gets \text{DH}(sk, pk')`$                          | Perform a Diffie-Hellman agreement between two ristretto255 keys, where $`K = sk \cdot pk' = sk' \cdot pk \in \mathbb{G}_{\mathrm{R255}}`$.                                                              |
```

cfm reviewed (changes requested):
Thanks, @rocodes!  Looks great; no substantive changes here, just proofreading for consistency.

rocodes committed:
docs: address review feedback

cfm commented on docs/protocol.md:
```diff
@@ -394,9 +397,8 @@ Then:
-| $`x \gets^{\$} \mathcal{E}_H`$[^8]                                                                                                                                                                            |                                 |                                                |
-| $`X \gets g^x`$                                                                                                                                                                                               |                                 |                                                |
-| $`Z \gets (pk_{R,i}^{fetch})^x`$                                                                                                                                                                              |                                 |                                                |
+| $`(x, X) \gets^{\$} \text{Ristretto255.KGen}()`$[^8]                                                                                                                                                          |                                 |                                                |
```
Yeah, you're totally right.  Unlike the `SD-APKE` and `SD-PKE` abstractions from the manuscript, the `Fetch` abstraction I suggested isn't useful if it's competing with the unabstracted primitive it interacts with!  I'm sorry for the detour.

rocodes committed:
docs: use ristretto255.KGen() instead of Fetch.KGen(). This is to
provide consistent notation wherever we are using ristretto255 (hint and
fetching key).

cfm reviewed (approved):
Thanks, @rocodes—including for workshopping it with me.
